lib/tasks/scheduler.rake

```
@@ -0,0 +1,22 @@
+desc 'Search nexmo delivered message'
+task :search_nexmo_delivered_message => :environment do
+  pending_messages = NexmoDeliveryReceipt.where(body: nil).order(id: :asc).limit(10)
+
+  if pending_messages.any?
+    client = Nexmo::Client.new(key: Rails.application.secrets.nexmo_api_key, secret: Rails.application.secrets.nexmo_api_secret)
+    response = client.search_messages(pending_messages.map(&:message_id))
+
+    response["items"].each do |item|
+      if message = NexmoDeliveryReceipt.find_by(message_id: item["message-id"])
+        message.body = item["body"]
+        if message["type"] == "MT"
+          message.date_received = message["date-received"]
+          message.final_status = message["final-status"]
+          message.date_closed = message["date-closed"]
+          message.latency = message["latency"]
+        end
+        message.save
```

`save!` will abort this task. but with `save`. We can silently pass this failure to make sure we can process all the message in this request. if we failed to save the response, the message body will still empty, we will search the message next time.